packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -366,7 +379,19 @@ class RenderEditable extends RenderBox {
 
   @override
   double computeMaxIntrinsicHeight(double width) {
-    return _preferredLineHeight * maxLines;
+    if (maxLines != null)
+      return _preferredLineHeight * maxLines;
+    if (width == double.INFINITY) {
+      final String text = _textPainter.text.toPlainText();
+      int lines = 1;
+      for (int index = 0; index < text.length; index += 1) {
+        if (text.codeUnitAt(index) == 0x0A) // count explicit line breaks
```

Comment:
After looking at this more and more carefully, I have concluded as follows:

RenderEditable's computeMinIntrinsicHeight was wrong. It should have been doing the same as computeMaxIntrinsicHeight (the method above). This is because it used to return that its minimum height was one line, but that's bogus since the object tries to be maxLines high and clips if it is less than that, and the docs say the min intrinsic height should be the height without clipping.

RenderEditable's max intrinsic height is correct. It makes sense that it's at this level because this is some weird behaviour that TextPainter and RenderParagraph don't have or want.